### PR TITLE
color.cc constexpr std::sqrt is not ISO C++

### PR DESCRIPTION
**Pull request by ahundt**
```
Semantic Issue Group
/Users/athundt/source/cartographer/cartographer/io/color.cc:32:17: Constexpr variable 'kGoldenRatioConjugate' must be initialized by a constant expression
```

https://stackoverflow.com/questions/8622256/in-c11-is-sqrt-defined-as-constexpr

### CODE REVIEW

File: cartographer/io/color.cc [hunk 1]
```diff
@@ -69,6 +68,7 @@ Color GetColor(int id) {
   CHECK_GE(id, 0);
   // Uniform color sampling using the golden ratio from
   // http://martin.ankerl.com/2009/12/09/how-to-create-random-colors-programmatically/
+  float kGoldenRatioConjugate = (std::sqrt(5.f) - 1.f) / 2.f;
```
Comment (SirVer) [inline cartographer/io/color.cc]: const float golden_ration_conjugate

Comment (ahundt) [inline cartographer/io/color.cc]: maybe static const? then it should only be calculated once

Or could add something like [this compile time implementation](https://baptiste-wicht.com/posts/2014/07/compile-integer-square-roots-at-compile-time-in-cpp.html) to math.h:

```c++
#define MID(a, b) ((a+b)/2)
#define POW(a) (a*a)

template<std::size_t res, std::size_t l = 1, std::size_t r = res>
struct ct_sqrt;

template<std::size_t res, std::size_t r>
struct ct_sqrt<res, r, r> : std::integral_constant<std::size_t, r> {};

template <std::size_t res, std::size_t l, std::size_t r>
struct ct_sqrt : std::integral_constant<std::size_t, ct_sqrt<res,
        (POW(MID(r, l)) >= res ? l : MID(r, l)+1),
        (POW(MID(r, l)) >= res ? MID(r, l) : r)>::value> {};
```

Comment (ahundt) [inline cartographer/io/color.cc]: Or even better just use the constant provided by boost:
http://www.boost.org/doc/libs/1_63_0/libs/math/doc/html/math_toolkit/constants.html

Comment (macmason) [inline cartographer/io/color.cc]: Or evaluate it in python, and then write the result down: `0.6180339887498949f`

Comment (ahundt) [inline cartographer/io/color.cc]: ah I resent you and your practical solutions @macmason. :-)

**Pull request merged**